web/client/utils/styleparser/StyleParserUtils.js

```diff
@@ -917,7 +917,8 @@ export const drawIcons = (geoStylerStyle, options) => {
     }, []);
     const marks = symbolizers.filter(({ kind }) => kind === 'Mark');
     const icons = symbolizers.filter(({ kind }) => kind === 'Icon');
-    return loadFontAwesome()
+    const shouldLoadFontAwesomeForIcons =  (geoStylerStyle.loadFontAwesomeForIcons && icons.length > 0) ? loadFontAwesome() : Promise.resolve();
+    return shouldLoadFontAwesomeForIcons
```

Comment:
what I asked was something different. 

I imagined something like this with **loadFontAwesomeForIcons** be put at root level in localConfig

```suggestion
    const loadFontAwesomeForIcons = getConfigProp("loadFontAwesomeForIcons");
    // if undefined or true it will load it to preserve previous behaviour
        const loadingPromise =  isNil(loadFontAwesomeForIcons) || loadFontAwesomeForIcons ? loadFontAwesome() : Promise.resolve();
    return loadingPromise 
```

so technically all other changes are not needed